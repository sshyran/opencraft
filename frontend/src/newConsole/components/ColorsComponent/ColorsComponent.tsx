import { InstancesModel } from 'console/models';
import * as React from 'react';
import { ConsolePage } from 'newConsole/components';
// import messages from './displayMessages';
import './styles.scss';
import { ThemeColors } from 'console/components';

interface State {}
interface ActionProps {
  updateThemeFieldValue: Function;
}
interface StateProps extends InstancesModel {}
interface Props extends StateProps, ActionProps {
  history: {
    goBack: Function;
  };
}

export class ColorsComponent extends React.PureComponent<
  Props,
  State
> {
  public render() {
    return(
      <ConsolePage
        contentLoading={this.props.loading}
        goBack={this.props.history.goBack}
        showSideBarEditComponent
      >
        <h1 className="edit-heading">Colors</h1>
        <ThemeColors></ThemeColors>
      </ConsolePage>
    )
  }

}