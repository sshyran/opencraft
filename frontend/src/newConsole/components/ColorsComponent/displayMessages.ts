const messages = {
    colors: {
        defaultMessage: 'Colors',
        description: 'Title test for colors edit page',
    },
};

export default messages;
