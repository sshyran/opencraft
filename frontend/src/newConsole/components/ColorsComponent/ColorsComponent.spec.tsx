import React from 'react';
import { setupComponentForTesting } from "utils/testing";
import { ColorsComponent } from './ColorsComponent';

it('renders without crashing', () => {
    const tree = setupComponentForTesting(<ColorsComponent />).toJSON();
    expect(tree).toMatchSnapshot();
});
