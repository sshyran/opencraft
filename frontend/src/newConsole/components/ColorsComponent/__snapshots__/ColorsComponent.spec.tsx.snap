// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders without crashing 1`] = `
<div
  className="console-page"
>
  <div
    className="d-flex justify-content-center align-middle email-activation-alert"
  >
    <i
      className="alert-icon fas fa-exclamation-triangle fa-xs"
    />
    One of your email addresses is pending confirmation. Please confirm it to begin managing your Open edX instance.
  </div>
  <div
    className="new-console-page-container"
  >
    <div
      className="new-console-page-content row"
    >
      <div
        className="container-fluid"
      >
        <div
          className="justify-content-center row"
        >
          <div
            className="col-md-3"
          >
            <button
              className="back-button btn btn-link btn-sm"
              disabled={false}
              onClick={[Function]}
              type="button"
            >
              <span>
                <i
                  aria-hidden="true"
                  className="fa fa-angle-left sm"
                />
              </span>
              <span>
                Back
              </span>
            </button>
            <h1
              className="edit-heading"
            >
              Colors
            </h1>
            <div
              className="row"
            >
              <div
                className="side-buttons col"
              >
                <div
                  className="color-input form-group"
                >
                  <label
                    className="form-label"
                  >
                    Main Color
                  </label>
                  <div
                    className="row"
                  >
                    <input
                      className="input-field-color form-control"
                      disabled={true}
                      name="mainColor"
                      onClick={[Function]}
                      readOnly={true}
                      value="#444444"
                    />
                    <div
                      className="input-field-preview"
                      style={
                        Object {
                          "backgroundColor": "#444444",
                        }
                      }
                    />
                    <div
                      className="info-icon"
                      onBlur={[Function]}
                      onFocus={[Function]}
                      onMouseOut={[Function]}
                      onMouseOver={[Function]}
                    >
                      <i
                        className="fas fa-info-circle"
                      />
                    </div>
                  </div>
                  <p>
                    <button
                      className="reset-value"
                      onClick={[Function]}
                      type="button"
                    >
                      Reset
                    </button>
                  </p>
                </div>
                <div
                  className="color-input form-group"
                >
                  <label
                    className="form-label"
                  >
                    Link Color
                  </label>
                  <div
                    className="row"
                  >
                    <input
                      className="input-field-color form-control"
                      disabled={true}
                      name="linkColor"
                      onClick={[Function]}
                      readOnly={true}
                      value="#FFAAFF"
                    />
                    <div
                      className="input-field-preview"
                      style={
                        Object {
                          "backgroundColor": "#FFAAFF",
                        }
                      }
                    />
                    <div
                      className="info-icon"
                      onBlur={[Function]}
                      onFocus={[Function]}
                      onMouseOut={[Function]}
                      onMouseOver={[Function]}
                    >
                      <i
                        className="fas fa-info-circle"
                      />
                    </div>
                  </div>
                  <p>
                    <button
                      className="reset-value"
                      onClick={[Function]}
                      type="button"
                    >
                      Reset
                    </button>
                  </p>
                </div>
              </div>
            </div>
          </div>
          <div
            className="col-md-7"
          >
            <div
              className="preview-box card"
            />
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;
